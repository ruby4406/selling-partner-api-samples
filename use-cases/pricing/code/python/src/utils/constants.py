from enum import Enum

# Region configuration
NA_REGION_CODE = "NA"
SP_API_NA_ENDPOINT = "https://sellingpartnerapi-na.amazon.com"
EU_REGION_CODE = "EU"
SP_API_EU_ENDPOINT = "https://sellingpartnerapi-eu.amazon.com"
FE_REGION_CODE = "FE"
SP_API_FE_ENDPOINT = "https://sellingpartnerapi-fe.amazon.com"

# Sandbox configuration
SANDBOX_REGIONS = ['NA_SANDBOX', 'EU_SANDBOX', 'FE_SANDBOX']
NA_SANDBOX_REGION_CODE = "NA_SANDBOX"
SP_API_NA_SANDBOX_ENDPOINT = "https://sandbox.sellingpartnerapi-na.amazon.com"
EU_SANDBOX_REGION_CODE = "EU_SANDBOX"
SP_API_EU_SANDBOX_ENDPOINT = "https://sandbox.sellingpartnerapi-eu.amazon.com"
FE_SANDBOX_REGION_CODE = "FE_SANDBOX"
SP_API_FE_SANDBOX_ENDPOINT = "https://sandbox.sellingpartnerapi-fe.amazon.com"


# Region to Endpoint mapping
VALID_SP_API_REGION_CONFIG = {
    NA_REGION_CODE: SP_API_NA_ENDPOINT,
    NA_SANDBOX_REGION_CODE: SP_API_NA_SANDBOX_ENDPOINT,
    EU_REGION_CODE: SP_API_EU_ENDPOINT,
    EU_SANDBOX_REGION_CODE: SP_API_EU_SANDBOX_ENDPOINT,
    FE_REGION_CODE: SP_API_FE_ENDPOINT,
    FE_SANDBOX_REGION_CODE: SP_API_FE_SANDBOX_ENDPOINT
}

# MarketplaceId to Region mapping
MARKETPLACE_ID_TO_REGION_CODE_MAPPING = {
    "A1234567895435": [NA_REGION_CODE, NA_SANDBOX_REGION_CODE],   # CA
    "A1234567895435":  [NA_REGION_CODE, NA_SANDBOX_REGION_CODE],   # US
    "A1234567895435": [NA_REGION_CODE, NA_SANDBOX_REGION_CODE],   # MX
    "A1234567895435": [NA_REGION_CODE, NA_SANDBOX_REGION_CODE],   # BR
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # ES
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # UK
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # FR
    "A1234567895435":  [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # BE
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # NL
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # DE
    "A1234567895435":  [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # IT
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # SE
    "A1234567895435":  [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # ZA
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # PL
    "A1234567895435":  [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # EG
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # TR
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # SA
    "A1234567895435": [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # AE
    "A1234567895435":  [EU_REGION_CODE, EU_SANDBOX_REGION_CODE],   # IN
    "A1234567895435": [FE_REGION_CODE, FE_SANDBOX_REGION_CODE],   # SG
    "A1234567895435": [FE_REGION_CODE, FE_SANDBOX_REGION_CODE],   # AU
    "A1234567895435": [FE_REGION_CODE, FE_SANDBOX_REGION_CODE]    # JP
}


# Login With Amazon Configuration
LWA_ENDPOINT = "https://api.amazon.com/auth/o2/token"


# Lambda Environment Variables
SP_API_APP_CREDENTIALS_SECRET_ARN_ENV_VARIABLE = "SP_API_APP_CREDENTIALS_SECRET_ARN"
SELLER_ITEMS_TABLE_NAME_ENV_VARIABLE = "SELLER_ITEMS_TABLE_NAME"


# Generic Lambda Input Parameters
REGION_CODE_KEY_NAME = "RegionCode"
REFRESH_TOKEN_KEY_NAME = "RefreshToken"


# State Machine Attributes
STATE_MACHINE_ARN_ENV_VARIABLE = "STATE_MACHINE_ARN"
STATE_MACHINE_KEY_REFRESH_TOKEN = "refreshToken"
STATE_MACHINE_KEY_REGION_CODE = "regionCode"
STATE_MACHINE_KEY_ASIN = "asin"
STATE_MACHINE_KEY_CREDENTIALS = "credentials"
STATE_MACHINE_KEY_OFFER_TYPE = "offerChangeType"
STATE_MACHINE_KEY_BUY_BOX = "buyBox"
STATE_MACHINE_KEY_BUY_BOX_CONDITION = "condition"
STATE_MACHINE_KEY_BUY_BOX_PRICE = "price"
STATE_MACHINE_KEY_SELLER = "seller"
STATE_MACHINE_KEY_SELLER_ID = "sellerId"
STATE_MACHINE_KEY_OFFERS = "offers"
STATE_MACHINE_KEY_SHIPPING_PRICE = "shippingPrice"
STATE_MACHINE_KEY_LISTING_PRICE = "listingPrice"
STATE_MACHINE_KEY_IS_BUY_BOX_WINNER = "isBuyBoxWinner"
STATE_MACHINE_KEY_MARKETPLACE_ID = "marketplaceId"
STATE_MACHINE_KEY_IS_FULFILLED_BY_AMAZON = "isFulfilledByAmazon"
STATE_MACHINE_KEY_COMPETITIVE_PRICE = "competitivePriceThreshold"
STATE_MACHINE_KEY_REFERENCE_PRICE = "referencePrice"
STATE_MACHINE_KEY_AVERAGE_SELLING_PRICE = "averageSellingPrice"
STATE_MACHINE_KEY_SELLER_OFFER = "sellerOffer"


# AWS Attributes
AWS_STEP_FUNCTIONS_CLIENT_NAME = "stepfunctions"
AWS_STEP_FUNCTIONS_ARN_NAME = "executionArn"
REFRESH_TOKEN_ARN_ENV_VARIABLE = "REFRESH_TOKEN"
REGION_CODE_ARN_ENV_VARIABLE = "REGION_CODE"
SQS_QUEUE_ARN_ENV_VARIABLE = "SQS_QUEUE_ARN"
SECRETS_MANAGER_CLIENT_NAME = "secretsmanager"


# Notifications Configurations
NOTIFICATION_TYPE_AOC = "ANY_OFFER_CHANGED"
NOTIFICATION_TYPE_PH = "PRICING_HEALTH"
LWA_NOTIFICATIONS_SCOPE = "sellingpartnerapi::notifications"


# Notification Attributes
NOTIFICATION_KEY_NOTIFICATION_TYPE = "notificationType"
NOTIFICATION_KEY_OFFER_CHANGE_TYPE = "OfferChangeType"
NOTIFICATION_KEY_OFFER_CHANGE_TRIGGER = "OfferChangeTrigger"
NOTIFICATION_KEY_ASIN = "ASIN"
NOTIFICATION_KEY_MARKETPLACE_ID = "MarketplaceId"
NOTIFICATION_KEY_SUMMARY = "Summary"
NOTIFICATION_KEY_BUY_BOX_PRICES = "BuyBoxPrices"
NOTIFICATION_BUY_BOX_PRICES_ITEM = 0
NOTIFICATION_KEY_ITEM_CONDITION = "Condition"
NOTIFICATION_KEY_LANDED_PRICE = "LandedPrice"
NOTIFICATION_KEY_OFFERS = "Offers"
NOTIFICATION_KEY_SELLER_ID = "SellerId"
NOTIFICATION_KEY_LISTING_PRICE = "ListingPrice"
NOTIFICATION_KEY_SHIPPING_PRICE = "Shipping"
NOTIFICATION_KEY_IS_BUY_BOX_WINNER = "IsBuyBoxWinner"
NOTIFICATION_KEY_IS_FULFILLED_BY_AMAZON = "IsFulfilledByAmazon"


# PRICING_HEALTH Attributes
NOTIFICATION_KEY_PH_OFFER_CHANGE_TRIGGER = "offerChangeTrigger"
NOTIFICATION_KEY_PH_ASIN = "asin"
NOTIFICATION_KEY_PH_MARKETPLACE_ID = "marketplaceId"
NOTIFICATION_KEY_PH_SUMMARY = "summary"
NOTIFICATION_KEY_PH_BUY_BOX_PRICES = "buyBoxPrices"
NOTIFICATION_BUY_PH_BOX_PRICES_ITEM = 0
NOTIFICATION_KEY_PH_ITEM_CONDITION = "itemCondition"
NOTIFICATION_KEY_PH_CONDITION = "condition"
NOTIFICATION_KEY_PH_LANDED_PRICE = "landedPrice"
NOTIFICATION_KEY_PH_MERCHANT_OFFER = "merchantOffer"
NOTIFICATION_KEY_PH_SELLER_ID = "sellerId"
NOTIFICATION_KEY_PH_LISTING_PRICE = "listingPrice"
NOTIFICATION_KEY_PH_SHIPPING_PRICE = "shipping"
NOTIFICATION_KEY_PH_REFERENCE_PRICE = "referencePrice"
NOTIFICATION_KEY_PH_AVERAGE_SELLING_PRICE = "averageSellingPrice"
NOTIFICATION_KEY_PH_COMPETITIVE_PRICE_THRESHOLD = "competitivePriceThreshold"
NOTIFICATION_KEY_PH_RETAIL_OFFER_PRICE = "retailOfferPrice"
NOTIFICATION_KEY_PH_MSRP_PRICE = "msrpPrice"
NOTIFICATION_KEY_PH_FULFILLMENT_TYPE = "fulfillmentType"
NOTIFICATION_KEY_PH_MFN_TYPE = "MFN"


# API Type
LISTINGS_API_TYPE = "listings"
PRICING_API_TYPE = "pricing"
NOTIFICATIONS_API_TYPE = "notifications"


# DynamoDB Attributes
SELLER_TABLE_HASH_KEY_NAME = "SKU"
AWS_DYNAMO_DB_CLIENT_KEY_NAME = "dynamodb"
SELLER_TABLE_NAME_ENV_VARIABLE = "SELLER_ITEMS_TABLE_NAME"
SELLER_TABLE_ASIN_KEY_NAME = "ASIN"
SELLER_TABLE_CONDITION_KEY_NAME = "Condition"
SELLER_TABLE_MARKETPLACE_KEY_NAME = "MarketplaceId"
SELLER_TABLE_SELLER_ID_KEY_NAME = "SellerId"
SELLER_TABLE_IS_FULFILLED_BY_AMAZON_KEY_NAME = "IsFulfilledByAmazon"
SELLER_TABLE_THRESHOLD_KEY_NAME = "MinThreshold"
SELLER_TABLE_PRICE_CHANGE_RULE_KEY_NAME = "PriceChangeRule"
SELLER_TABLE_PRICE_CHANGE_AMOUNT_KEY_NAME = "PriceChangeRuleAmount"
SELLER_TABLE_USE_COMPETITIVE_PRICE = "UseCompetitivePrice"


# Price Change Enum Value
class PriceChangeRule(Enum):
    PERCENTAGE = "PERCENTAGE"
    FIXED = "FIXED"
