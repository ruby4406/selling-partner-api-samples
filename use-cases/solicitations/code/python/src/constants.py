from typing import Dict

class Constants:
    # Region configuration
    NA_REGION_CODE = 'NA'
    SP_API_NA_ENDPOINT = 'https://sellingpartnerapi-na.amazon.com'
    SP_API_NA_SANDBOX_ENDPOINT = 'https://sandbox.sellingpartnerapi-na.amazon.com'
    EU_REGION_CODE = 'EU'
    SP_API_EU_ENDPOINT = 'https://sellingpartnerapi-eu.amazon.com'
    SP_API_EU_SANDBOX_ENDPOINT = 'https://sandbox.sellingpartnerapi-eu.amazon.com'
    FE_REGION_CODE = 'FE'
    SP_API_FE_ENDPOINT = 'https://sellingpartnerapi-fe.amazon.com'
    SP_API_FE_SANDBOX_ENDPOINT = 'https://sandbox.sellingpartnerapi-fe.amazon.com'

    # Region to endpoint mapping
    REGION_ENDPOINT_MAPPING = {
        NA_REGION_CODE: (SP_API_NA_ENDPOINT, SP_API_NA_SANDBOX_ENDPOINT),
        EU_REGION_CODE: (SP_API_EU_ENDPOINT, SP_API_EU_SANDBOX_ENDPOINT),
        FE_REGION_CODE: (SP_API_FE_ENDPOINT, SP_API_FE_SANDBOX_ENDPOINT)
    }

    # Marketplace Id to region code mapping
    MARKETPLACE_ID_TO_REGION_CODE_MAPPING = {
        'A2EUQ1WTGCTBG2': 'NA', #CA
        'ATVPDKIKX0DER': 'NA', #US
        'A1AM78C64UM0Y8': 'NA', #MX
        'A2Q3Y263D00KWC': 'NA', #BR
        'A1RKKUPIHCS9HS': 'EU', #ES
        'A1F83G8C2ARO7P': 'EU', #UK
        'A13V1IB3VIYZZH': 'EU', #FR
        'AMEN7PMS3EDWL': 'EU', #BE
        'A1805IZSGTT6HS': 'EU', #NL
        'A1PA6795UKMFR9': 'EU', #DE
        'APJ6JRA9NG5V4': 'EU', #IT
        'A2NODRKZP88ZB9': 'EU', #SE
        'AE08WJ6YKNBMC': 'EU', #ZA
        'A1C3SOZRARQ6R3': 'EU', #PL
        'ARBP9OOSHTCHU': 'EU', #EG
        'A33AVAJ2PDY3EV': 'EU', #TR
        'A17E79C6D8DWNP': 'EU', #SA
        'A2VIGQ35RCS4UG': 'EU', #AE
        'A21TJRUUN4KGV': 'EU', #IN
        'A19VAU5U5O7RUS': 'FE', #SG
        'A39IBJ37TRP1C6': 'FE', #AU
        'A1VC38T7YXB528': 'FE', #JP
    }

    # Login With Amazon configuration
    LWA_ENDPOINT = 'https://api.amazon.com/auth/o2/token'
    LWA_NOTIFICATIONS_SCOPE = 'sellingpartnerapi::notifications'

    # Notifications configuration
    NOTIFICATION_TYPE_ORDER_CHANGE = 'ORDER_CHANGE'
    ORDER_CHANGE_NOTIFICATION_STATUS_SHIPPED = 'Shipped'

    # Lambda environment variables
    REFRESH_TOKEN_ENV_VARIABLE = 'REFRESH_TOKEN'
    SP_API_APP_CREDENTIALS_SECRET_ARN_ENV_VARIABLE = 'SP_API_APP_CREDENTIALS_SECRET_ARN'
    SOLICITATIONS_SCHEDULER_ROLE_ARN_ENV_VARIABLE = 'SOLICITATIONS_SCHEDULER_ROLE_ARN'
    SOLICITATIONS_STATE_MACHINE_ARN_ENV_VARIABLE = 'SOLICITATIONS_STATE_MACHINE_ARN'
    SQS_QUEUE_ARN_ENV_VARIABLE = 'SQS_QUEUE_ARN'

    # Generic Lambda input parameters
    REGION_CODE_KEY_NAME = 'RegionCode'
    REFRESH_TOKEN_KEY_NAME = 'RefreshToken'

    # Solicitations API constants
    ACTION_PRODUCT_REVIEW_SELLER_FEEDBACK = 'productReviewAndSellerFeedback'