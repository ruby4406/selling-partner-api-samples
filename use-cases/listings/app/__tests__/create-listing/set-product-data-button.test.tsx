import { act, screen, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { NextIntlProvider } from "next-intl";
import React from "react";
import fetch from "jest-fetch-mock";
import translations from "@/app/internationalization/translations/en-US.json";
import { US_LOCALE } from "@/app/constants/global";
import { mockResolveFetchResponse } from "@/app/test-utils/mock-fetch";
import userEvent from "@testing-library/user-event";
import AlertProvider from "@/app/context/alert-context-provider";
import SetProductDataButton from "@/app/components/set-product-data-button";

const MOCK_VARIATION_JSON_PROPERTY = "parentage_level";
const MOCK_PARENT_JSON_PROPERTY_VALUE = {
  value: "parent",
};
const MOCK_BUTTON_ID = "test-save-product-data-button";
const MOCK_BUTTON_TEXT = "test-button";
const MOCK_BUTTON_ALERT = "test-alert";
let MOCK_LISTING = {};
const MOCK_SET_DATA = (data: any) => {
  return data;
};

function renderSetProductDataButton() {
  return render(
    <NextIntlProvider locale={US_LOCALE} messages={translations}>
      <AlertProvider>
        <SetProductDataButton
          property={{
            propertyName: MOCK_VARIATION_JSON_PROPERTY,
            propertyValue: MOCK_PARENT_JSON_PROPERTY_VALUE,
          }}
          buttonId={MOCK_BUTTON_ID}
          buttonText={MOCK_BUTTON_TEXT}
          buttonAlert={MOCK_BUTTON_ALERT}
          existingData={MOCK_LISTING}
          setData={MOCK_SET_DATA}
        />
      </AlertProvider>
    </NextIntlProvider>,
  );
}

describe("Test Set Product Data Button", () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  test("Test alert displays on button click.", async () => {
    mockResolveFetchResponse(200, {
      data: {},
    });

    let components = await act(async () => {
      const { asFragment, getByLabelText, getByRole, getByTestId } =
        renderSetProductDataButton();
      return { asFragment, getByLabelText, getByRole, getByTestId };
    });

    await act(async () => {
      await userEvent.click(
        screen.getByTestId("test-save-product-data-button"),
      );
    });

    await waitFor(() =>
      expect(screen.queryByTestId("product-data-set")).toBeVisible(),
    );
    expect(components.asFragment()).toMatchSnapshot();
  });
});
