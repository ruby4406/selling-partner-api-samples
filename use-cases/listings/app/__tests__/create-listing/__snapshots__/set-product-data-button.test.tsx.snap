// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Set Product Data Button Test alert displays on button click. 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-ethv1g-MuiPaper-root-MuiAlert-root"
            data-testid="product-data-set"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              test-alert
            </div>
            <div
              class="MuiAlert-action css-ki1hdl-MuiAlert-action"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-testid="test-save-product-data-button"
    id="test-save-product-data-button"
    tabindex="0"
    type="button"
  >
    test-button
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
